server/app.py

```diff
@@ -21,9 +21,34 @@
 
 gs = GroundStation(socketio)
 
+logger = logging.getLogger("main")
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s [%(levelname)-8s]  %(message)s")
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.WARNING)
+console_handler.setFormatter(formatter)
+
+file_handler = logging.FileHandler("log.txt", mode="w")
+file_handler.setLevel(logging.INFO)
+file_handler.setFormatter(formatter)
+
+debug_file_handler = logging.FileHandler("debug_log.txt", mode="w")
+debug_file_handler.setLevel(logging.DEBUG)
+debug_file_handler.setFormatter(formatter)
+
+logger.addHandler(console_handler)
+logger.addHandler(file_handler)
+logger.addHandler(debug_file_handler)
+
+print("LOGGING STARTED")
+logger.info("LOGGING STARTED")
+
 
 @app.errorhandler(Exception)
 def handle_error(e):
+    logger.error(type(e).__name__)
+    logger.info("Traceback of %s : ", type(e).__name__, exc_info=e)
```

Comment:
There is also a [`logger.exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) method which does this combined. I understand that there would be a rational for doing it like this though, I just wanted to make sure you considered it.

Note that there are other flask handlers which use this method, for if you decide to change it.

Reply:
> There is also a logger.exception method which does this combined. 


I know that, and yes I have considered it. My main rationale on choosing this option is solely because of the different log levels. At the `ERROR` level, I only want to print the fact that an Exception occured, because printing full traceback would make it unmanageable. However, I still wanted to log the full traceback for debugging purposes. Therefore, I logged this at the `INFO` level. Basically, here is what different things will see:

console
```
TIMESTAMP [ERROR   ]  ZeroDivisionError
```

log.txt & debug_log.txt
```
TIMESTAMP [ERROR   ]  ZeroDivisionError
TIMESTAMP [INFO    ]  Traceback (most recent call last):
more...
```

So yes, this was intentional.